### PR TITLE
[ci] Fix iOS client build

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Noticed it fails (see https://github.com/expo/expo/pull/9528).

# How

Since it's the upload part which fails and the error says "missing credentials" I figured we may want to ensure env variables are passed to the command.

# Test Plan

Let's see what CI thinks.

### CODE REVIEW

**Comment by sjchmiela** [comment]
I got to a point where the build succeeds, but it fails to upload not due to missing credentials, but to invalid credentials:
```
2020-08-04T17:19:09.4304530Z Build 2.16.0 is already defined in versions endpoint.
2020-08-04T17:19:09.4304830Z The new build would be uploaded onto the same URL.
2020-08-04T17:19:09.4306290Z Uploading 2.16.0 build
...
2020-08-04T17:19:14.2060760Z a ./GoogleMaps.bundle/GMSCoreResources.bundle/ja.lproj/GMSCore.strings
2020-08-04T17:19:14.2066770Z a ./GoogleMaps.bundle/GMSCoreResources.bundle/el.lproj/GMSCore.strings
2020-08-04T17:19:14.2071250Z a ./GoogleMaps.bundle/GMSCoreResources.bundle/ar.lproj/GMSCore.strings
2020-08-04T17:19:14.2075350Z a ./GoogleMaps.bundle/GMSCoreResources.bundle/en_AU.lproj/GMSCore.strings
2020-08-04T17:19:14.2079130Z a ./GoogleMaps.bundle/GMSCoreResources.bundle/he.lproj/GMSCore.strings
2020-08-04T17:19:14.2083430Z a ./GoogleMaps.bundle/GMSCoreResources.bundle/de.lproj/GMSCore.strings
2020-08-04T17:19:14.2100190Z a ./_CodeSignature/CodeResources
2020-08-04T17:19:14.3714650Z There was an error running client-build command: Access Denied
2020-08-04T17:19:14.3714840Z 
2020-08-04T17:19:14.3784060Z ##[error]Process completed with exit code 1.
```

If not AWS credentials from Github secrets vault then… which ones? Please, help, @nicknovitski! 🙏

**Comment by sjchmiela** [comment]
Ha! Using AWS credentials from a corresponding Android build job it passed!

File: .github/workflows/client-ios.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@ jobs:
           fi
       - run: echo "::add-path::$(pwd)/bin"
       - run: expotools ios-generate-dynamic-macros
+      - run: expotools ios-generate-dynamic-macros
```
Comment (bbarthec) [inline .github/workflows/client-ios.yml:50]:
```suggestion
```
Do we really need another invocation of this process here? 🤔

Comment (sjchmiela) [reply .github/workflows/client-ios.yml:50]: It's a hotfix for https://github.com/expo/expo/issues/9547

Comment (bbarthec) [reply .github/workflows/client-ios.yml:50]: Maybe some comment then?
At first glace it seems as a additional unnecessary line 😉

Comment (sjchmiela) [reply .github/workflows/client-ios.yml:50]: I intend to remove this line before merging the PR to fix one bug at at time, but thanks for reminding me to make it clear this is not part of the PR. 👍

File: .github/workflows/client-ios.yml [hunk 2]
```diff
@@ -63,6 +63,9 @@ jobs:
           bundle install --jobs 4 --retry 3
       - run: expotools client-build --platform ios
         timeout-minutes: 60
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
Comment (tsapeta) [inline .github/workflows/client-ios.yml:68]: Wouldn't it be better to add this env to the entire job instead of both steps?

Comment (sjchmiela) [reply .github/workflows/client-ios.yml:68]: I followed the convention set by the `docs` workflow

https://github.com/expo/expo/blob/63f388bed1978890a2e57dab03c0871c6643c706/.github/workflows/docs.yml#L67-L71

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**